l2geth/rollup/client.go

```diff
@@ -90,11 +84,13 @@ type decoded struct {
 type RollupClient interface {
 	GetEnqueue(index uint64) (*types.Transaction, error)
 	GetLatestEnqueue() (*types.Transaction, error)
-	GetTransaction(index uint64) (*types.Transaction, error)
-	GetLatestTransaction() (*types.Transaction, error)
-	GetEthContext(index uint64) (*EthContext, error)
+	GetTransaction(uint64, string) (*types.Transaction, error)
+	GetLatestTransaction(string) (*types.Transaction, error)
```

Comment:
The `string` argument here is used as a query param sent to the DTL to specify the data source - `l1` or `l2`, see https://github.com/ethereum-optimism/optimism/pull/479

Reply:
This should probably be an enum type instead of a string?

Reply:
This was updated to use an enum type